meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc

```diff
@@ -592,9 +591,39 @@ STM32MP_DT_FILES_DK:remove:stm32mp15-disco = "stm32mp157f-dk2"
 SOTA_CLIENT_FEATURES:append:stm32mp15-disco = " ubootenv"
 IMAGE_FSTYPES:append:stm32mp15-disco = " ext4"
 IMAGE_FSTYPES:remove:sota:stm32mp15-disco = "ext4"
-LMP_FLASHLAYOUT:stm32mp15-disco ?= "FlashLayout_sd_stm32mp157c-dk2-optee.tsv"
-LMP_FLASHLAYOUT_TEMPLATE:stm32mp15-disco ?= "FlashLayout_sd_stm32mp157c-dk2-optee.tsv.in"
 LMP_FLASHLAYOUT_BOARD_NAME:stm32mp15-disco = "stm32mp157c-dk2"
+FLASHLAYOUT_PARTITION_DEVICE:stm32mp15-disco = "mmc0"
+# USB FSBP
+FLASHLAYOUT_PARTITION_OFFSET:${STM32MP_FSBL_PROGAMMER_NAME}:stm32mp15-disco = "0x0"
+FLASHLAYOUT_PARTITION_DEVICE:${STM32MP_FSBL_PROGAMMER_NAME}:stm32mp15-disco = "none"
+FLASHLAYOUT_PARTITION_BIN2LOAD:${STM32MP_FSBL_PROGAMMER_NAME}:stm32mp15-disco = "tf-a-${LMP_FLASHLAYOUT_BOARD_NAME}-usb${STM32_BOOTIMAGE_SUFFIX}.stm32"
+# USB FIP
+FLASHLAYOUT_PARTITION_OFFSET:${STM32MP_SSBL_PROGAMMER_NAME}:stm32mp15-disco = "0x0"
+FLASHLAYOUT_PARTITION_BIN2LOAD:${STM32MP_SSBL_PROGAMMER_NAME}:stm32mp15-disco = "fip-${LMP_FLASHLAYOUT_BOARD_NAME}-optee.bin"
+FLASHLAYOUT_PARTITION_DEVICE:${STM32MP_SSBL_PROGAMMER_NAME}:stm32mp15-disco = "none"
+# FSBL1
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:${STM32MP_FSBL1_NAME}:stm32mp15-disco = "PD"
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:${STM32MP_FSBL1_NAME}:stm32mp15-disco = "0x00004400"
+FLASHLAYOUT_PARTITION_BIN2LOAD:${STM32MP_FSBL1_NAME}:stm32mp15-disco = "tf-a-${LMP_FLASHLAYOUT_BOARD_NAME}-sdcard${STM32_BOOTIMAGE_SUFFIX}.stm32"
+# FSBL2
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:${STM32MP_FSBL2_NAME}:stm32mp15-disco = "PD"
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:${STM32MP_FSBL2_NAME}:stm32mp15-disco = "0x00044400"
+FLASHLAYOUT_PARTITION_BIN2LOAD:${STM32MP_FSBL2_NAME}:stm32mp15-disco = "tf-a-${LMP_FLASHLAYOUT_BOARD_NAME}-sdcard${STM32_BOOTIMAGE_SUFFIX}.stm32"
+# FIP-A
+FLASHLAYOUT_PARTITION_ENABLE:sdcard:fip-a:stm32mp15-disco = "PD"
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:fip-a:stm32mp15-disco = "0x00084400"
+FLASHLAYOUT_PARTITION_BIN2LOAD:fip-a:stm32mp15-disco = "fip-${LMP_FLASHLAYOUT_BOARD_NAME}-optee.bin"
+# FIP-B
+FLASHLAYOUT_PARTITION_ENABLE:sdcard:fip-b:stm32mp15-disco = "PD"
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:fip-b:stm32mp15-disco = "0x00484400"
+# U-Boot env
+FLASHLAYOUT_PARTITION_BIN2LOAD:fip-b:stm32mp15-disco = "fip-${LMP_FLASHLAYOUT_BOARD_NAME}-optee.bin"
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:u-boot-env = "0x00900000"
+FLASHLAYOUT_PARTITION_ENABLE:sdcard:u-boot-env = "PED"
+# ROOTFS
+FLASHLAYOUT_PARTITION_OFFSET:sdcard:${STM32MP_ROOTFS_LABEL} = "0x00910000"
+FLASHLAYOUT_PARTITION_ENABLE:sdcard:${STM32MP_ROOTFS_LABEL} = "P"
+FLASHLAYOUT_PARTITION_ENABLE:sdcard:${STM32MP_ROOTFS_LABEL} = "${MFGTOOL_FLASH_IMAGE}-${MACHINE}.ext4"
```

Comment:
There are two `FLASHLAYOUT_PARTITION_ENABLE:sdcard:${STM32MP_ROOTFS_LABEL}` assignments so first one is replaced by the second.